desktop/cmp/modalsupport/ModalSupportModel.ts

```diff
@@ -6,30 +6,27 @@
  */
 
 import {HoistModel} from '@xh/hoist/core';
-import {ModalSupportOptions} from '@xh/hoist/desktop/cmp/panel';
+import {ModalSupportOptions} from '@xh/hoist/desktop/cmp/modalsupport';
 import '@xh/hoist/desktop/register';
-import {bindable, makeObservable} from '@xh/hoist/mobx';
+import {observable, makeObservable, action} from '@xh/hoist/mobx';
 import {createObservableRef} from '@xh/hoist/utils/react';
 
 /**
  * Core Model for a ModalSupport component.
  * This model will place its component's child in 1 of 2 managed DOM nodes (either modal or inline)
- * @private
  */
 export class ModalSupportModel extends HoistModel {
     xhImpl = true;
 
-    @bindable isModal = false;
+    @observable
+    isModal: boolean = false;
+    options: ModalSupportOptions;
 
-    inlineRef = createObservableRef();
-    modalRef = createObservableRef();
-    hostNode;
-    options;
+    inlineRef = createObservableRef<HTMLElement>();
+    modalRef = createObservableRef<HTMLElement>();
+    hostNode: HTMLElement;
 
-    /**
-     * @param {ModalSupportOptions|Object} opts
-     */
-    constructor(opts = new ModalSupportOptions()) {
+    constructor(opts: Partial<ModalSupportOptions> = new ModalSupportOptions()) {
```

Comment:
This is interesting! Need to read a bit more about `Partial` and how that works...